### PR TITLE
add `NotLink` error

### LINKED ISSUE #9872: 'readLink' fails on non-symlink file
### Summary

When running the std.os.readLink function on a non-link path on Linux, an unreachable place of code is reached.

### Setup
1. WSL or Linux environment
2. Zig 0.8.1

### Reproduction
1. Choose a non-link file in linux.
```
var buffer: [2 << 20]u8 = undefined;
const result = try std.fs.cwd().readLink(link, buffer[0..]);
```
result:
```
/home/foobar/zig/lib/std/os.zig:2559:19: 0x20a017 in std.os.readlinkatZ (readlink)
        EINVAL => unreachable,    
```
### Analysis
The system call likely returns EINVAL if a non-link file is link resolved. The code at readlinkatZ doesn't expect this value as the switch case in unreachable.

### Possible solution
Add a 'NotALink' error and return it when receiving EINVAL from the system.

### PR DESCRIPTION
**Pull request by AliChraghi**
fixes #9872

### CODE REVIEW

File: lib/std/zig/system.zig [hunk 1]
```diff
@@ -612,6 +612,7 @@ pub const NativeTargetInfo = struct {
             error.AccessDenied => return error.GnuLibCVersionUnavailable,
             error.FileSystem => return error.FileSystem,
             error.SymLinkLoop => return error.SymLinkLoop,
+            error.NotLink => return error.NotLink,
```
Comment (squeek502) [inline lib/std/zig/system.zig:615]: I think the quickest fix would be to just do:

```suggestion
            error.NotLink => return error.GnuLibCVersionUnavailable,
```

Comment (AliChraghi) [reply lib/std/zig/system.zig:615]: thx, but I can not build the zig and I have to see the result in CI which leads to a waste of resources

Comment (AliChraghi) [reply lib/std/zig/system.zig]: @squeek502 thank you. build is successful!

**Comment by squeek502** [comment]
Looks like you might have committed line ending changes to all files accidentally with your last commit.

**Comment by AliChraghi** [comment]
yeah :smile: I added lib/ directory instead lib/std.

**Review by marler8997 — APPROVED** [review]
Looks good to me now

**Comment by vrischmann** [comment]
The linked issue is affecting Fedora 35 users, essentially making the `zig` command unusable because it tries to readlink `/lib64/libc.so.6` which is a normal file.

I can confirm that with this PR zig works again.

**Comment by andrewrk** [comment]
Thank you @AliChraghi and reviewers :+1:

**Pull request merged**